lib/nerves_init_gadget/ssh_console.ex

```diff
@@ -0,0 +1,63 @@
+defmodule Nerves.InitGadget.SSHConsole do
+  @moduledoc """
+  SSH IEx console.
+  """
+  use GenServer
+
+  @doc """
+  Since ctrl-c is intercepted in default Nerves config,
+  We need to be able to exit the shell somehow.
+  """
+  def shell_exit, do: GenServer.call(__MODULE__, :ssh_exit)
+
+  @doc false
+  def start_link(%{ssh_console_port: nil}), do: :ignore
+
+  def start_link(opts) do
+    GenServer.start_link(__MODULE__, [opts], name: __MODULE__)
+  end
+
+  def init([opts]) do
+    ssh = start_ssh(opts)
+    {:ok, %{ssh: ssh, opts: opts}}
+  end
+
+  def terminate(_, %{ssh: ssh}) do
+    :ssh.stop_daemon(ssh)
+  end
+
+  def handle_call(:ssh_exit, from, %{ssh: ssh} = state) do
+    # Reply to avoid crashing the call.
+    GenServer.reply(from, :ok)
+    :ok = :ssh.stop_daemon(ssh)
```

Comment:
If you have more than one ssh connection, this ends up killing it too. I'm thinking that if you document that you should send `<enter>`, then `~.` that we don't need this function anyway.

Reply:
How did you find `<enter>` then `~.`??

Reply:
That's an `ssh` thing. Check out the escape code section of the man page.